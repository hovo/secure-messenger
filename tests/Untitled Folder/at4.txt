add_group(1,"group1")
add_group(2,"group2")
add_group(3,"group3")
add_user(1,"hovo")
add_user(2,"vahe")
register_user(1,1)
register_user(1,2)
register_user(1,3)
register_user(2,1)
register_user(2,3)
list_users
list_groups
send_message(1,1,"aasdsa")
send_message(2,1, "asdasdasdasdasd")
send_message(1,2, "acxxcx")
read_message(1,1)
read_message(2,2)
read_message(3,3)
list_new_messages(1)
list_new_messages(2)
list_new_messages(3)
list_old_messages(3)
list_old_messages(2)
list_old_messages(1)
