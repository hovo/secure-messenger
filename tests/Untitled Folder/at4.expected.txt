  0:  OK

->add_group(1,"group1")
  1:  OK
  Users:
  Groups:
      1->group1
  Registrations:
  All messages:
  Message state:

->add_group(2,"group2")
  2:  OK
  Users:
  Groups:
      1->group1
      2->group2
  Registrations:
  All messages:
  Message state:

->add_group(3,"group3")
  3:  OK
  Users:
  Groups:
      1->group1
      2->group2
      3->group3
  Registrations:
  All messages:
  Message state:

->add_user(1,"hovo")
  4:  OK
  Users:
      1->hovo
  Groups:
      1->group1
      2->group2
      3->group3
  Registrations:
  All messages:
  Message state:

->add_user(2,"vahe")
  5:  OK
  Users:
      1->hovo
      2->vahe
  Groups:
      1->group1
      2->group2
      3->group3
  Registrations:
  All messages:
  Message state:

->register_user(1,1)
  6:  OK
  Users:
      1->hovo
      2->vahe
  Groups:
      1->group1
      2->group2
      3->group3
  Registrations:
      [1, hovo]->{1->group1}
  All messages:
  Message state:

->register_user(1,2)
  7:  OK
  Users:
      1->hovo
      2->vahe
  Groups:
      1->group1
      2->group2
      3->group3
  Registrations:
      [1, hovo]->{1->group1, 2->group2}
  All messages:
  Message state:

->register_user(1,3)
  8:  OK
  Users:
      1->hovo
      2->vahe
  Groups:
      1->group1
      2->group2
      3->group3
  Registrations:
      [1, hovo]->{1->group1, 2->group2, 3->group3}
  All messages:
  Message state:

->register_user(2,1)
  9:  OK
  Users:
      1->hovo
      2->vahe
  Groups:
      1->group1
      2->group2
      3->group3
  Registrations:
      [1, hovo]->{1->group1, 2->group2, 3->group3}
      [2, vahe]->{1->group1}
  All messages:
  Message state:

->register_user(2,3)
  10:  OK
  Users:
      1->hovo
      2->vahe
  Groups:
      1->group1
      2->group2
      3->group3
  Registrations:
      [1, hovo]->{1->group1, 2->group2, 3->group3}
      [2, vahe]->{1->group1, 3->group3}
  All messages:
  Message state:

->list_users
  11:  OK
  1->hovo
  2->vahe

->list_groups
  12:  OK
  1->group1
  2->group2
  3->group3

->send_message(1,1,"aasdsa")
  13:  OK
  Users:
      1->hovo
      2->vahe
  Groups:
      1->group1
      2->group2
      3->group3
  Registrations:
      [1, hovo]->{1->group1, 2->group2, 3->group3}
      [2, vahe]->{1->group1, 3->group3}
  All messages:
      1->[sender: 1, group: 1, content: "aasdsa"]
  Message state:
      (1, 1)->read
      (2, 1)->unread

->send_message(2,1,"asdasdasdasdasd")
  14:  OK
  Users:
      1->hovo
      2->vahe
  Groups:
      1->group1
      2->group2
      3->group3
  Registrations:
      [1, hovo]->{1->group1, 2->group2, 3->group3}
      [2, vahe]->{1->group1, 3->group3}
  All messages:
      1->[sender: 1, group: 1, content: "aasdsa"]
      2->[sender: 2, group: 1, content: "asdasdasdasdasd"]
  Message state:
      (1, 1)->read
      (2, 1)->unread
      (1, 2)->unread
      (2, 2)->read

->send_message(1,2,"acxxcx")
  15:  OK
  Users:
      1->hovo
      2->vahe
  Groups:
      1->group1
      2->group2
      3->group3
  Registrations:
      [1, hovo]->{1->group1, 2->group2, 3->group3}
      [2, vahe]->{1->group1, 3->group3}
  All messages:
      1->[sender: 1, group: 1, content: "aasdsa"]
      2->[sender: 2, group: 1, content: "asdasdasdasdasd"]
      3->[sender: 1, group: 2, content: "acxxcx"]
  Message state:
      (1, 1)->read
      (2, 1)->unread
      (1, 2)->unread
      (2, 2)->read
      (1, 3)->read
      (2, 3)->unavailable

->read_message(1,1)
  16:  ERROR 
  Message has already been read. See `list_old_messages'.

->read_message(2,2)
  17:  ERROR 
  Message has already been read. See `list_old_messages'.

->read_message(3,3)
  18:  ERROR 
  User with this ID does not exist.

->list_new_messages(1)
  19:  OK
  New/unread messages for user [1, hovo]:
      2->[sender: 2, group: 1, content: "asdasdasdasdasd"]

->list_new_messages(2)
  20:  OK
  New/unread messages for user [2, vahe]:
      1->[sender: 1, group: 1, content: "aasdsa"]

->list_new_messages(3)
  21:  ERROR 
  User with this ID does not exist.

->list_old_messages(3)
  22:  ERROR 
  User with this ID does not exist.

->list_old_messages(2)
  23:  OK
  Old/read messages for user [2, vahe]:
      2->[sender: 2, group: 1, content: "asdasdasdasdasd"]

->list_old_messages(1)
  24:  OK
  Old/read messages for user [1, hovo]:
      1->[sender: 1, group: 1, content: "aasdsa"]
      3->[sender: 1, group: 2, content: "acxxcx"]

